src/main/java/bio/terra/service/iam/sam/SamIam.java

```diff
@@ -78,6 +79,23 @@ public boolean isAuthorized(
         IamResourceType iamResourceType,
         String resourceId,
         IamAction action) {
+
+        SamRetry samRetry = new SamRetry(samConfig);
+        while (true) {
```

Comment:
It feels like this could be DRY'd up. What do you think about adding an instance method on SamRetry that can take a callable for the inside of the try block here? I.e.

```java
samRetry.perform(() -> isAuthorizedInner(userReq, iamResourceType, resourceId, action));
```

Reply:
This is _sooooo_ cool! The prettiest use of lambdas I've seen, if I do say so myself :)

Two benefits from this approach. First, the retry while loops go away. Second, the ApiException handling can be done _inside_ the `perform`, so it DRYs up all of the repetitive
```
try {
   . . .
} catch (ApiException ex) {
   throw convertSAMExToDataRepoEx(ex);
}
```
in each Inner function. I had to make a custom `@FunctionalInterface` for SAM that specified the throw.